api/v1beta1/clusterdeployment_types.go

```diff
@@ -72,6 +72,21 @@ type ClusterDeploymentSpec struct {
 	// PropagateCredentials indicates whether credentials should be propagated
 	// for use by CCM (Cloud Controller Manager).
 	PropagateCredentials bool `json:"propagateCredentials,omitempty"`
+
+	// IPAMClaim defines IP Address Management (IPAM) requirements for the cluster.
+	// It can either reference an existing IPAM claim or specify an inline claim.
+	IPAMClaim ClusterIPAMClaimType `json:"ipamClaim,omitempty"`
+}
+
+// ClusterIPAMClaimType represents the IPAM claim configuration for a cluster deployment.
+// It allows referencing an existing claim or defining a new one inline.
+type ClusterIPAMClaimType struct {
+	// ClusterIPAMClaimRef is the name of an existing ClusterIPAMClaim resource to use.
+	ClusterIPAMClaimRef string `json:"ref,omitempty"`
```

Comment:
I am a little bit confused about the ClusterDeployment and ClusterIPAMClaim dependencies.

The ClusterDeployment has a reference to ClusterIPAMClaim, ClusterIPAMClaim has a reference to ClusterDeployment. 

What happens if there is a ClusterIPAMClaim `claim1` with `spec.clusterDeploymentRef = foo` and a ClusterDeployment `bar` that references this claim (`spec.ipamClaim.ref = claim1`)?

Also, from the current implementation, two or more ClusterDeployments can reference one IPAMClaim. But the IPAMClaim has a single `clusterDeploymentRef` field.

I think there is a circular dependency that we should avoid. Or am I misunderstanding something?